cmd/syncthing/gui.go

```diff
@@ -968,10 +969,17 @@ func (s *apiService) getRandomString(w http.ResponseWriter, r *http.Request) {
 func (s *apiService) getDBIgnores(w http.ResponseWriter, r *http.Request) {
 	qs := r.URL.Query()
 
-	ignores, patterns, err := s.model.GetIgnores(qs.Get("folder"))
+	folder := qs.Get("folder")
+	ignores, patterns, err := s.model.GetIgnores(folder)
 	if err != nil {
-		http.Error(w, err.Error(), 500)
-		return
+		matcher := ignore.New(false)
+		path := filepath.Join(s.cfg.Folders()[folder].Path(), ".stignore")
```

Comment:
This looks dangerous. If the folder doesn't exist, `s.cfg.Folders()[folder].Path()` is the empty string and `.stignore` ends up in the current directory, whatever that happens to be for the running syncthing instance... It's probably a "file not found" here but the `POST` case below will create the file if it's missing, as far as I can see.